sqlmesh/dbt/context.py

```diff
@@ -62,7 +62,7 @@ class DbtContext:
     def dialect(self) -> str:
         if not self.target:
             raise SQLMeshError("Target must be configured before calling the dialect property.")
-        return self.target.dialect
+        return self.sqlmesh_config.model_defaults.dialect or self.target.dialect
```

Comment:
i changed this because i was trying to run a client's snowflake dbt in duckdb but was unable to do so.

Reply:
@treysp @crericha

Reply:
Transpiling is currently not supported via the dbt adapter. If we want to support, there is a larger change set then these two changes required. Basically, there are some places in the dbt adapter where we need to know the target DWH dialect and other places that need to know the model's dialect. Right now, they are treated as the same thing.

Reply:
well for basic changes like this, it works fine, just trying to see column level lineage

Reply:
There are other basic changes that would hit an issue, so its more so happenstance that you didn't hit one of the other places that need to be changed. I'm on call next week and can add transpiling to the dbt adapter if this is a priority.

Reply:
no it’s not but this was enough to unblock me. is it incorrect?

Reply:
Yes, I believe it is incorrect. The context dialect is sometimes used for the target DWH.

Reply:
I think we'd want context to have a default_model_dialect method and then find the places in the adapter where that should be used instead of the target dialect.

Reply:
As a bonus, we could add dialect as a config property and allow individual models to override the default.

Reply:
i'll remove these changes for now